lib/elixir/lib/enum.ex

```diff
@@ -2584,8 +2584,9 @@ defmodule Enum do
   Enumerates the `enumerable`, by removing the elements for which
   function `fun` returned duplicate items.
 
-  The function `fun` maps every element to a term which is used to
-  determine if two elements are duplicates.
+  The function `fun` maps every element to a term and then compares the
```

Comment:
I would say this is a bit confusing though as it's not `fun` that "compares the terms for each element [...]". I have some trouble thinking of a way to improve the current documentation which seems clear to me, but I can try:

> The function `fun` maps every element to a term. `uniq_by/2` compares such terms to determine what corresponding elements in the `enumerable` are duplicates.

Thoughts?